pyvex/block.py

```diff
@@ -69,14 +71,67 @@ def __init__(self, data, mem_addr, arch, max_inst=None, max_bytes=None, bytes_of
         self._size = None
 
         if data is not None:
-            lift(self, data, max_bytes, max_inst, bytes_offset, opt_level, traceflags)
+            lift(self, arch, mem_addr, data, max_bytes, max_inst, bytes_offset, opt_level, traceflags)
+
+    @staticmethod
+    def empty_block(arch, addr, statements=None, nxt=None, tyenv=None, jumpkind=None, direct_next=None, size=None):
+        block = IRSB(None, addr, arch)
+        block._set_attributes(statements, nxt, tyenv, jumpkind, direct_next)
+        return block
+
+    def copy(self):
+        return copy.deepcopy(self)
+
+    def extend(self, extendwith):
+        conversion_dict = { }
+        invalid_vals = (0xffffffff, -1)
+
+        def convert_tmp(tmp):
+            if tmp not in conversion_dict:
+                tmp_type = extendwith.tyenv.lookup(tmp)
+                conversion_dict[tmp] = self.tyenv.add(tmp_type)
+            return conversion_dict[tmp]
+
+        def convert_expr(expr):
+            if isinstance(expr, RdTmp):
```

Comment:
A super brief comment with what these are for (so we can maintain them later) would help a lot.